### ISSUE TITLE
Publication of a paper about matscipy

### ISSUE THREAD

**Issue by pgrigorev** [libAtoms/matscipy#108]
So far, while citing `matscipy` in our papers I had to use a web citation to the repository that always feels a bit _ad-hoc_. Thus I was thinking that it would be nice to have a paper published to be able to formally cite the code. Moreover, it can help all of us to get more formal recognition of our efforts. This issue is essentially to discuss how we can achieve this. 

After a short discussion about it wit @jameskermode we decided that [Journal of Open Source Software](https://joss.theoj.org/) looks like a nice option. It requires only a very short (250-1000 words) high level description of the software, but the review is more focused on the software itself: installation, tests coverage and documentation.  You can find more details on [submission requirements](https://joss.readthedocs.io/en/latest/submitting.html#submission-requirements) and [reviewer checklist](https://joss.readthedocs.io/en/latest/review_checklist.html) pages. 

We should host the `paper.md` file as a part of the repository, that can be in a short lived branch that is never merged to the main repository. Then everyone can add a small bit with the description of the parts of the code that they developed together with more general description. I am happy to volunteer to do the bookkeeping for this branch and the submission itself.

It seems to me that in the current shape `matscipy` already passes most of the requirements and writing a short summary should not take too long. However, before starting working on the submission it makes sense to understand if there are parts of `matscipy` that require a significant rework before the submission. I would suggest using the discussion on this issue to provide your opinions. On my site, I probably should enable building the documentation for the `dislocation.py` module as well as to think about solving a dependency on `atomman` package. 

I am happy to discuss other options and suggestions as well.

**Comment by pastewka** [comment]
I think this is a good idea. @prs513rosewood has published in this journal before and has commented positively on it.

**Comment by jameskermode** [comment]
Sounds good to me, happy to be involved and to contribute to the writing.

**Comment by pastewka** [comment]
Same here.

**Comment by pastewka** [comment]
PS: I don't think that there are parts that require significant rework. However, it is to be understood that the code (and API) can change over time. We should also adopt a more stringent versioning policy and release more often.

**Comment by pastewka** [comment]
That said, one thing I wanted to do at some point is to make our C-accelerated neighbor list 100% compatible with what is now implemented in the main ASE branch in pure Python.

**Comment by pgrigorev** [comment]
Oh, very impressive quick reactions, thank you! It seems to be the case that there is nothing really that stops us to go for it. The dependency on `atomman` and the compatibility of neighbour lists  with `ASE` looks more like possible enhancements. We can think about platform where we can list and discuss it. I see there is something like `projects` in the repository. Can this be a good place?

As for the submission, I have just created a new branch [108-JOSS-sbmission](https://github.com/libAtoms/matscipy/tree/108-JOSS-submission) for it. I will make the first draft there and keep you updated!

**Comment by jameskermode** [comment]
Does JOSS have a suggested method for determining authorship? I think we should be as inclusive as possible and invite everyone who has contributed code to be an author.

**Comment by pastewka** [comment]
I agree about being as inclusive as possible here.

**Comment by threich** [comment]
The IWM folks that are involved in matscipy (Sliding_P barostat implementation and the OPLS-LAMMPS interface) would also  be happy to contribute, if you consider our contributions to matscipy significant enough.

**Comment by pgrigorev** [comment]
> Does JOSS have a suggested method for determining authorship? I think we should be as inclusive as possible and invite everyone who has contributed code to be an author.

They have a [short guidance](https://joss.readthedocs.io/en/latest/submitting.html#authorship)

_Purely financial (such as being named on an award) and organizational (such as general supervision of a research group) contributions are not considered sufficient for co-authorship of JOSS submissions, but active project direction and other forms of non-code contributions are. The authors themselves assume responsibility for deciding who should be credited with co-authorship, and co-authors must always agree to be listed. In addition, co-authors agree to be accountable for all aspects of the work, and to notify JOSS if any retraction or correction of mistakes are needed after publication._

It means it is up to us to decide the authorship. And I agree we should be as inclusive as possible.

**Comment by pgrigorev** [comment]
I have just merged few changes in the submission branch [108-JOSS-submission](https://github.com/libAtoms/matscipy/tree/108-JOSS-submission). I will repeat the summary of what I have done here. My idea is that anyone who wants to contribute can put there changes to this branch and then we use it fro submission.  

I have added a bit of text and the suggested structure of the paper. I think it would make sense to describe each part of `matscipy` in the paper since they are quite different from each other. I have added a short paragraph for the `dislocation.py` module including a citation of a paper that were published thanks to it. Please feel free to do the same to for other parts. Also add more parts if I forgot something important. Keep in mind that we have to stay under 1000 words though.

We could think about one figure with an illustration of the atomistic systems for each case to make the paper a bit less boring. 

As we agreed before we should be as inclusive as possible so please extend the author list.

I also added an github action that builds the draft pdf as an artifact. 
_The PDF is available via the Actions tab in your project and click on the latest workflow run. The zip archive file (including the `paper.pdf`) is listed in the run’s Artifacts section._

Alternatively you can use docker to build the draft:
```
docker run --rm \
    --volume $PWD/paper:/data \
    --user $(id -u):$(id -g) \
    --env JOURNAL=joss \
    openjournals/inara
```
You have to run it in the `matscipy` folder. it assumes `paper.md` is in the `paper` folder.

**Comment by pgrigorev** [comment]
I have just merged a pull request from @threich with the description of `pressurecoupling.py` - great work! It is nice to see that we are progressing. If anyone wants to contribute, but does not want or does not have time to fiddle with branches and pull requests, he can just post the text here as a comment and I will be happy to add it to the draft. Same for the citations. Thank you!

**Comment by jameskermode** [comment]
@pgrigorev sorry it's taken a while to get around to this, but I've just added a short paragraph on fracture mechanics tools with a few references, and started a new section at the end of the paper where we can collect general non-domain-specific tools. The example I had in mind was our neighbour list, which is becoming quite widely used, so I added that, but no doubt there are others. Let me know if these changes are ok or if you need more detail.

**Comment by pgrigorev** [comment]
@jameskermode this is fantastic, thanks a lot! No worries for the delay, I think the end of the year is a busy period for everyone. It is a great idea to add non domain specific tools as well. We are now at around 600 words, so we still have some room (we have to stay below 1000 words). Once we have two missing sections added we can fine tune the level of details to make sure it is readable. It is really great that we are progressing. 

In the meantime we could think about a single figure with an example illustration for each section. My idea is just to have a set of nice atomistic configurations, just to make it a bit more cheerful that plain text. Something like cracks on the [James page](https://warwick.ac.uk/fac/sci/eng/people/james_kermode). What do you think?

**Comment by pgrigorev** [comment]
I think @prs513rosewood has a perfect illustration for the tribology section on [his page](https://frerot.net/research/) as well. :)

**Comment by prs513rosewood** [comment]
I've been thinking about the figures, and made a quick proof-of-concept figure for a crack tip.
![crack_mockup](https://user-images.githubusercontent.com/328212/216335813-28ebfd99-0e41-40f3-a585-c8e76896d8ab.png)

Data is from the 2d crack example simulation in LAMMPS. I don't want to put figures in the paper that are unrelated to matscipy, so if you have atomic configurations (preferably in a format supported by Ovito) that are related to matscipy I can try to make them into a figure similar to the one above (made in Blender).

**Comment by pgrigorev** [comment]
This is fantastic! The figure looks amazing. I tried to use blender at some point as well Here I attached a `xyz` file for a junction dislocation in BCC, it might look good as well. (I had to compress it to zip otherwise github wont attach it) 
[BCC_junction_100_edge_dislocation.zip](https://github.com/libAtoms/matscipy/files/10569903/BCC_junction_100_edge_dislocation.zip)

**Comment by pgrigorev** [comment]
I was also thinking to set up a little deadline for everyone who wants to contribute to the paper. This will allow us to keep the momentum and stay a bit more organised. Since we have already advanced quite a lot, lets aim for a submission by the end of February if that is ok for everyone.

**Comment by aklemenz** [comment]
Submission by the end of February should be possible for the IWM colleagues. We still have to finalize a few things, but should be ready in one to two weeks.

**Comment by pgrigorev** [comment]
This is great! Thanks a lot.

**Comment by jameskermode** [comment]
Agree with the deadline, will do what I can to help

**Comment by prs513rosewood** [comment]
On the Freiburg side this week is the last week of classes, schedules should clear up after that. I'd just like to remind everyone that the paper is not the only thing that needs work. The review check-list for JOSS (see e.g. [here](https://github.com/openjournals/joss-reviews/issues/2121) for the review of a code of mine) includes:

> **Documentation > Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems)."
> **Documentation > Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

It's good that matscipy provides examples (are they still working?), but I think the documentation on what one can do, or what one might want to do, with matscipy (ie *not* API documentation) is lacking.

> This is fantastic! The figure looks amazing. I tried to use blender at some point as well Here I attached a `xyz` file for a junction dislocation in BCC, it might look good as well. (I had to compress it to zip otherwise github wont attach it) [BCC_junction_100_edge_dislocation.zip](https://github.com/libAtoms/matscipy/files/10569903/BCC_junction_100_edge_dislocation.zip)

I've made this, let me know if I can improve/change anything.

![dislocation_colors](https://user-images.githubusercontent.com/328212/217016263-0ec47b3a-d942-4153-bd3b-97134edfad36.png)

**Comment by pastewka** [comment]
I may not be able to meet the end of Feb deadline, but I agree that we should try.

Regarding examples, we should include them in the tests. (I.e. there should be a test whether examples run.)

**Comment by pgrigorev** [comment]
> It's good that matscipy provides examples (are they still working?), but I think the documentation on what one can do, or what one might want to do, with matscipy (ie not API documentation) is lacking.

Thanks for pointing this out. I planned to have a run through the and the documentation of the `dislocation` module.

> I've made this, let me know if I can improve/change anything.

Thanks a lot, it looks great. The angle and the coloring is a bit misleading though. Did you color it by hand? You could send me the blender scene file and I can play with it myself. 

>  Regarding examples, we should include them in the tests. (I.e. there should be a test whether examples run.)

I actually was going to generate few the examples for dislocation module from the tests, but I guess this should work as well :)

**Comment by prs513rosewood** [comment]
I colored based on positions. You should be able to tweak things in the shader node setup.
[bcc_dislocation.blend.zip](https://github.com/libAtoms/matscipy/files/10673220/bcc_dislocation.blend.zip)

**Comment by prs513rosewood** [comment]
I've created a PR #141 to collect changes to the documentation. Feel free to push directly to that PR.

**Comment by pgrigorev** [comment]
Hello everyone and thanks a lot for the recent progress on the paper and other related issues. I was recently notified that my application for a CNRS position (French way to recruit for permanent positions) was selected for an interview. This is good news, but it will be at the end of February. I wont be able to do a lot of work on the paper before the interview and will come back to it in March, so our soft deadline will be shifted a bit.

**Comment by pgrigorev** [comment]
I played a bit with the junction dislocation in blender. Here is the result with the extra plane of atoms colored by purple.
![dislocation](https://user-images.githubusercontent.com/28930707/224797482-8dd168cc-d92d-4575-a74b-4d683e5fdccc.png)

**Comment by jameskermode** [comment]
Also, @lakshenoy and @Fraser-Birks have been working on the numerical continuation flexible boundary condition fracture functionality, currently in the [adaptivecont](https://github.com/libAtoms/matscipy/tree/adaptivecont) branch but if we get that merged they could also jump on board with the paper.

**Comment by pastewka** [comment]
I have to apologize for being slow here. Will manage to look at paper next week.

**Comment by pgrigorev** [comment]
This is great, everyone is very welcome to join for sure. If the paper becomes to dense and heavy we could consider splitting it in few by the topics. I think the journal allows it. We can discuss this at final stages.

**Comment by pastewka** [comment]
Sorry, I am being slow here but want to look at the paper now. Where do I actually find it and how do I edit the text? Is in the 108-JOSS-submission? That looks like a normal repository to me.

**Comment by pgrigorev** [comment]
@pastewka I think the comment below answers your question. You can build the pdf yourself or use one of the actions to get it as an artifact. Please let me know if you still have difficulties obtaining it.

https://github.com/libAtoms/matscipy/issues/108#issuecomment-1283825723

**Comment by pastewka** [comment]
Okay thanks

**Comment by jameskermode** [comment]
@jgolebiowski We are writing a JOSS paper about matscipy and I am adding a short section about your MultiClusterForceMixingPotential class in `matscipy.calculators.mcfm`. Would you like to be included as a co-author? We are aiming to be as inclusive as possible.

**Comment by jameskermode** [comment]
@simonwengert We are writing a JOSS paper about matscipy and I am adding a short section about your committee models in `matscipy.calculators.committee`. Would you like to be included as a co-author? We are aiming to be as inclusive as possible.

**Comment by jameskermode** [comment]
@pgrigorev I will be visting Lars in Freiburg this coming week and we are keen to try to wrap up the paper then. If you have other changes you wanted to make please let us know!

**Comment by simonwengert** [comment]
> @simonwengert We are writing a JOSS paper about matscipy and I am adding a short section about your committee models in matscipy.calculators.committee. Would you like to be included as a co-author? We are aiming to be as inclusive as possible.

@jameskermode Nice to hear that you are planing to include my contributions in the paper. Sorry for not responding to your first message some time ago. I left for a job in industry which keeps me quite busy at the moment. I would be happy to be named as co-author, I can't tell how much time I will find for text-contributions, though.

**Comment by jgolebiowski** [comment]
> @jgolebiowski We are writing a JOSS paper about matscipy and I am adding a short section about your MultiClusterForceMixingPotential class in `matscipy.calculators.mcfm`. Would you like to be included as a co-author? We are aiming to be as inclusive as possible.

@jameskermode  Nice to hear that you are planing to write the paper about matscipy. I would be happy to be named as co-author, I will see if I have a more up-to-date version of `matscipy.calculators.mcfm` in my personal repositories and submit a PR if I can. Beyond that, please let me kow if I can contribute.

**Comment by pgrigorev** [comment]
> @pgrigorev I will be visting Lars in Freiburg this coming week and we are keen to try to wrap up the paper then. If you have other changes you wanted to make please let us know!

This is fantastic news! I have added small changes regarding the comment about an unclear sentence in the Plasticity section and also added a citation of my recent paper. 

As @prs513rosewood mentioned before, we should come up with a documentation with a consistent formatting (together with interactive examples?). It would be nice if you think about it during your visit. I think for the submission we can have something preliminary that can be improved during the review process. 

Also we already have a bit more than 2000 words and it is stated in the [submission instructions](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) that authors submitting papers significantly longer than 1000 words may be asked to reduce the length of their paper.

**Comment by jameskermode** [comment]
@lifelemons Hope all is well. We are writing a JOSS paper about matscipy. We are trying to be as inclusive as possible and include everyone who made substantive contributions to the code as an author? Would you be happy to be listed?

**Comment by pastewka** [comment]
@a-gola Hope you are doing okay. We are writing a JOSS paper about matscipy and we would like to include you as a coauthor for your past contribution to the EAM portion of it. Let me know if you are happy with that!

**Comment by pastewka** [comment]
@wgnoehring Hope you are doing okay. We are writing a JOSS paper about matscipy and we would like to include you as a coauthor for your past contribution to the EAM portion of it. Let me know if you are happy with that!

**Comment by jameskermode** [comment]
@Fraser-Birks and @lakshenoy we have added you both as authors to this paper in anticipation of your contributions to matscipy. If you have ORCID IDs then please add them, or let me know and I can do it (it would be a good time to setup one up already if not!).

**Comment by jameskermode** [comment]
@jgolebiowski  and @simonwengert: please could you send your email addresses to me so we can send you a close-to-final PDF of the paper for your review. `J DOT R DOT Kermode AT warwick DOT AC uk`

**Comment by jameskermode** [comment]
@pgrigorev We have added a new structure for the documentation which renders Jupyter notebooks for the examples. If you could add one or more notebooks for plasticity/dislocations and link them from `docs/applications/plasticity.rst` that would be great: you can see how it works in the neighbourlist one in `docs/tools/neighbour_list.ipynb` within the branch for PR #160. I will do one for fracture, and I think @prs513rosewood will do one for elasticity.

**Comment by jameskermode** [comment]
> @jgolebiowski and @simonwengert: please could you send your email addresses to me so we can send you a close-to-final PDF of the paper for your review. `J DOT R DOT Kermode AT warwick DOT AC uk`

Never mind, we found you both online
